clippy_lints/src/redundant_clone.rs

```diff
@@ -303,7 +302,7 @@ fn base_local_and_movability<'tcx>(
         mut projection,
     } = place.as_ref();
     if let PlaceBase::Local(local) = place_base {
-        while let Some(box Projection { base, elem }) = projection {
+        while let [base @ .., elem] = projection {
             projection = base;
```

Comment:
Remove this, is not needed anymore and change projection below on line 311 to base